src/main/java/org/jas/service/impl/LastfmServiceImpl.java

```diff
@@ -27,38 +27,45 @@
 import org.jas.action.ActionResult;
 import org.jas.service.LastfmService;
 import org.jas.service.LastFMCompleteService;
+import org.jas.service.config.LastFMConfig;
 
 import lombok.extern.slf4j.Slf4j;
 
-@Slf4j
-
-
 /**
  * @understands A class who completes metadata using LastFM service
  */
 
-@Service
-public class LastfmServiceImpl implements LastfmService {
-
-	@Autowired
-	private LastFMCompleteService completeService;
-
-
-	public synchronized ActionResult completeLastFM(Metadata metadata) {
-		try {
-			if (completeService.canLastFMHelpToComplete(metadata)) {
-				LastfmAlbum lastfmAlbum = completeService.getLastFM(metadata);
-				return completeService.isSomethingNew(lastfmAlbum, metadata);
-			} else {
-				return ActionResult.Ready;
-			}
-		} catch (MalformedURLException mfe) {
-			log.error(mfe.getMessage(), mfe);
-			return ActionResult.Error;
-		} catch (IOException ioe) {
-			log.error(ioe.getMessage(), ioe);
-			return ActionResult.Error;
-		}
-	}
-
-}
+ @Slf4j
+ @Service
+ public class LastfmServiceImpl implements LastfmService {
+ 
+	 @Autowired
+	 private LastFMCompleteService completeService;
+ 
+	 public synchronized ActionResult completeLastFM(Metadata metadata) {
+		 try {
+			 // Retrieve LastFM API Key and Secret from the environment
+			 String apiKey = LastFMConfig.getLastFMKey();
+			 String apiSecret = LastFMConfig.getLastFMSecret();
+ 
```

Comment:
You are reading here `KEY` and `SECRET` but not really using them or set, please notice we have two classes using `KEY` and `SECRET`
- `LastFMAlbumHelper`
- `AuthenticatorHelper`
Therefore we should read those values in those classes